apps/rsautl.c

```diff
@@ -208,64 +206,65 @@ int rsautl_main(int argc, char **argv)
     if (pkey == NULL)
         return 1;
 
-    rsa = EVP_PKEY_get1_RSA(pkey);
-    EVP_PKEY_free(pkey);
-
-    if (rsa == NULL) {
-        BIO_printf(bio_err, "Error getting RSA key\n");
-        ERR_print_errors(bio_err);
-        goto end;
-    }
-
     in = bio_open_default(infile, 'r', FORMAT_BINARY);
     if (in == NULL)
         goto end;
     out = bio_open_default(outfile, 'w', FORMAT_BINARY);
     if (out == NULL)
         goto end;
 
-    keysize = RSA_size(rsa);
+    keysize = EVP_PKEY_size(pkey);
 
     rsa_in = app_malloc(keysize * 2, "hold rsa key");
     rsa_out = app_malloc(keysize, "output rsa key");
+    rsa_outlen = keysize;
 
     /* Read the input data */
-    rsa_inlen = BIO_read(in, rsa_in, keysize * 2);
-    if (rsa_inlen < 0) {
+    rv = BIO_read(in, rsa_in, keysize * 2);
+    if (rv < 0) {
         BIO_printf(bio_err, "Error reading input Data\n");
         goto end;
     }
+    rsa_inlen = rv;
     if (rev) {
-        int i;
+        size_t i;
         unsigned char ctmp;
+
         for (i = 0; i < rsa_inlen / 2; i++) {
             ctmp = rsa_in[i];
             rsa_in[i] = rsa_in[rsa_inlen - 1 - i];
             rsa_in[rsa_inlen - 1 - i] = ctmp;
         }
     }
-    switch (rsa_mode) {
 
+    if ((ctx = EVP_PKEY_CTX_new_from_pkey(NULL, pkey, NULL)) == NULL)
+        goto end;
+
+    switch (rsa_mode) {
     case RSA_VERIFY:
-        rsa_outlen = RSA_public_decrypt(rsa_inlen, rsa_in, rsa_out, rsa, pad);
+        rv = EVP_PKEY_verify_recover_init(ctx)
+            && EVP_PKEY_CTX_set_rsa_padding(ctx, pad)
```

Comment:
Is the indentation here correct?  It looks one space shy.

If it is, the following `case` branchess are similar.

Reply:
Four space indent.  `case` statements are generally treated like labels, i.e. has negative indent applied.  Since they are inside curly braces, their indentation become +4-4, while the case bodies get +4 indentation.

Reply:
Do note that *inside parentheses*, however, we align with the opened parenthesis.  So if the add parentheses to that expression, we get this:

``` C
        rv = (EVP_PKEY_verify_recover_init(ctx)
              && EVP_PKEY_CTX_set_rsa_padding(ctx, pad)
              ...);
```

Reply:
Just to verify:

``` console
$ ./util/check-format.pl apps/rsautl.c 
apps/rsautl.c:73:function body length = 217 > 200 lines:int rsautl_main(int argc, char **argv)
1 (0 indentation, 0 directive nesting, 0 syntax, 0 whitespace, 1 length, 0 other) issues have been found by ./util/check-format.pl
```

And no, I'm not at all prepared to refactor `rsa_main`, that's hugely out of scope for this PR.

Reply:
I'd have thought the `&&` should line up since it's part of an expression.  I.e. one additional space indentation.

Not a show stopper.